mongo/datadog_checks/mongo/utils.py

```diff
@@ -27,7 +27,13 @@ def build_connection_string(hosts, scheme, username=None, password=None, databas
         netloc = host
 
     path_params = ""
-    query = urlencode(options or {})
+    url_options = dict(options)
+    if "username" in url_options:
+        url_options.pop("username")
+    if "password" in url_options:
+        url_options.pop("password")
```

Comment:
Quick question, what is the purposes of removing these params from the url_options?

Reply:
Username and password are valid options when initializing the mongo client, but they are not valid URI options: https://docs.mongodb.com/manual/reference/connection-string/#standard-connection-string-format. They are instead specified right before the host portion here: https://github.com/DataDog/integrations-core/blob/master/mongo/datadog_checks/mongo/utils.py#L21